cmd/fpga_admissionwebhook/README.md

```diff
@@ -1,97 +1,145 @@
 
-Verify that the `cfssl` and `jq` utilities are installed on your host.
-Run the `scripts/webhook-deploy.sh` script.
+Use the option `--mode` script option to deploy the webhook in orchestrated mode:
+
+```bash
+$ ./scripts/webhook-deploy.sh --mode orchestrated
 ```
-    $ cd $SRC
-    $ ./scripts/webhook-deploy.sh
-    Create secret including signed key/cert pair for the webhook
-    Creating certs in /tmp/tmp.XGTpddQBwP
-    certificatesigningrequest.certificates.k8s.io/intel-fpga-webhook-svc.default created
-    NAME                             AGE   REQUESTOR          CONDITION
-    intel-fpga-webhook-svc.default   0s    kubernetes-admin   Pending
-    certificatesigningrequest.certificates.k8s.io/intel-fpga-webhook-svc.default approved
-    secret/intel-fpga-webhook-certs created
-    Removing /tmp/tmp.XGTpddQBwP
-    Create FPGA CRDs
-    customresourcedefinition.apiextensions.k8s.io/acceleratorfunctions.fpga.intel.com created
-    customresourcedefinition.apiextensions.k8s.io/fpgaregions.fpga.intel.com created
-    fpgaregion.fpga.intel.com/arria10.dcp1.0 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.0-compress created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.0-nlb0 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.0-nlb3 created
-    fpgaregion.fpga.intel.com/arria10.dcp1.1 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.1-nlb0 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.1-nlb3 created
-    fpgaregion.fpga.intel.com/arria10.dcp1.2 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.2-nlb0 created
-    acceleratorfunction.fpga.intel.com/arria10.dcp1.2-nlb3 created
-    fpgaregion.fpga.intel.com/d5005 created
-    acceleratorfunction.fpga.intel.com/d5005-nlb0 created
-    acceleratorfunction.fpga.intel.com/d5005-nlb3 created
-    clusterrole.rbac.authorization.k8s.io/fpga-reader created
-    clusterrolebinding.rbac.authorization.k8s.io/default-fpga-reader created
-    Create webhook deployment
-    deployment.extensions/intel-fpga-webhook-deployment created
-    Create webhook service
-    service/intel-fpga-webhook-svc created
-    Register webhook
-    mutatingwebhookconfiguration.admissionregistration.k8s.io/fpga-mutator-webhook-cfg created
+
+The script needs the CA bundle used for signing certificate requests in your cluster.
+By default, the script fetches the bundle stored in the configmap
+`extension-apiserver-authentication`. However, your cluster may use a different signing
+certificate that is passed in the option `--cluster-signing-cert-file` to `kube-controller-manager`.
+In this case, you must point the script to the actual signing certificate as follows:
+
+```bash
+$ ./scripts/webhook-deploy.sh --ca-bundle-path /var/run/kubernetes/server-ca.crt
 ```
 
+# Mappings
+
+Requested FPGA resources are translated to AF resources. For example,
+`fpga.intel.com/arria10-nlb0` is translated to `fpga.intel.com/af-d8424dc4a4a3c413f89e433683f9040b`.
```

Comment:
1. Currently it's not enough to specify the model (arria10). Resource names must have the version of blue bitstream ("firmware") also, e.g. `fpga.intel.com/arria10.dcp1.1-nlb0`. Here `dcp1.1` is the version of the "firmware".

2. In orchestrated mode `fpga.intel.com/arria10.dcp1.1-nlb0` gets translated to `fpga.intel.com/region-9926ab6d6c925a68aabca7d84c545738` plus corresponding AF IDs are set in environment variables for the container. Then the scheduler schedules the pod to a node and the CRI-hook on the node loads the requested bitstream to a region before the container is started.